### PR TITLE
Add a cop to check for consistent method usage in feature specs.

### PR DESCRIPTION
**Pull request by rspeicher**

### CODE REVIEW

**Comment by tsigo** [comment]
I started down the path of making this `ConfigurableEnforcedStyle`, with `rspec` and `capybara` options.

If someone were to set `capybara`, we'd have to make sure to only check:

1. Files in `spec/features`
1. Files with `type: :feature` (and possibly just `:feature`?) metadata

I wasn't sure of the best way to handle those two conditions, so I'm open to feedback on making this configurable.

File: lib/rubocop/cop/rspec/feature_methods.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      #     end
+      #
+      #     it 'with OAuth' do
+      #       # ...
+      #     end
+      #   end
+      class FeatureMethods < Cop
+        MSG = 'Use `%s` instead of `%s`.'.freeze
+
+        # https://git.io/v7rLu
```
Comment (backus) [inline lib/rubocop/cop/rspec/feature_methods.rb]: If you hit `y` on your keyboard when visiting this link then github will give you a link to a sha that will never break. https://help.github.com/articles/getting-permanent-links-to-files/

Comment (tsigo) [inline lib/rubocop/cop/rspec/feature_methods.rb]: Sure, updated.

**Comment by bquorning** [comment]
I’ve been trying to find out *why* capybara provides these aliased methods. [The README](https://github.com/teamcapybara/capybara#the-dsl) just says that “Capybara also comes with a built in DSL for creating descriptive acceptance tests […] `feature` is in fact just an alias for `describe ..., :type => :feature`, `background` is an alias for `before`, `scenario` for `it`, and `given`/`given!` aliases for `let`/`let!`, respectively.” without any explanation for why this DSL should be preferable.

Also, I just want to note that these aliases are only available if Capybara (or rspec-rails) is loaded. Which puts this cop in the same category as `RSpec::FactoryGirl::DynamicAttributeDefinedStatically` as “not really an RSpec cop, but close”. Perhaps we should have an `RSpec::Capybara` module?

**Comment by rspeicher** [comment]
@bquorning I believe they're just syntactic sugar to make feature specs read more like [Cucumber](https://github.com/cucumber/cucumber/blob/master/docs/gherkin.md) features.

**Comment by Darhazer** [comment]
Yup, this should be in RSpec::Capybara module. There are few other things we could implement for capybara (like #378 for example)

**Comment by rspeicher** [comment]
Moved the copy to a new `Capybara` namespace.

File: spec/project/default_config_spec.rb [hunk 2]
```diff
@@ -6,10 +6,11 @@
   let(:cop_names) do
     namespaces = {
       'rspec' => 'RSpec',
+      'capybara' => 'Capybara',
       'factory_girl' => 'FactoryGirl'
     }
     glob = SpecHelper::ROOT.join('lib', 'rubocop', 'cop',
-                                 'rspec', '{,factory_girl/}*.rb')
+                                 'rspec', '{capybara/,,factory_girl/}*.rb')
```
Comment (rspeicher) [inline spec/project/default_config_spec.rb]: Requiring these two changes might get messy quickly if you anticipate adding more namespaces, but I went with the simplest thing for now.

**Comment by rspeicher** [comment]
Can't see what the build failures are since they're private.

**Comment by Darhazer** [comment]
@rspeicher you can run `rake` on your local machine to check the build. I think you should add `Capybara` to the namespaces in `ConfigFormatter` and fix the build_config script to include capybara
```
-cops = '{*.rb,factory_girl/*.rb}'
+cops = '{*.rb,factory_girl/*.rb,capybara/*.rb}'
```

**Comment by bquorning** [comment]
Ahh, I think you need to add `,capybara/*.rb` to bin/build_config for the `confirm_config` test to pass.

File: bin/build_config [hunk 3]
```diff
@@ -8,7 +8,7 @@ require 'rubocop-rspec'
 require 'rubocop/rspec/description_extractor'
 require 'rubocop/rspec/config_formatter'
 
-cops = '{*.rb,factory_girl/*.rb}'
+cops = '{*.rb,capybara/*.rb,factory_girl/*.rb}'
```
Comment (rspeicher) [inline bin/build_config]: This could be simplified to just `**/*.rb`.

**Comment by rspeicher** [comment]
Back to green, thanks y'all!

**Comment by rspeicher** [comment]
Rebased since 1.16.0 was released.

File: lib/rubocop/rspec/config_formatter.rb [hunk 4]
```diff
@@ -22,7 +22,7 @@ def unified_config
         cops.each_with_object(config.dup) do |cop, unified|
           unified[cop] = config.fetch(cop)
             .merge(descriptions.fetch(cop))
-            .merge('StyleGuide' => STYLE_GUIDE_BASE_URL + cop)
+            .merge('StyleGuide' => STYLE_GUIDE_BASE_URL + cop.sub('RSpec/', ''))
```
Comment (rspeicher) [inline lib/rubocop/rspec/config_formatter.rb]: Would you prefer an anchored regex here?

**Comment by Darhazer** [comment]
I think it's ready for merge. @backus final words?

**Comment by backus** [comment]
LGTM. @rspeicher or @Darhazer (whoever does the merge) can you squash down the commits?

**Pull request merged**